merkle/common.go

```diff
@@ -24,3 +24,10 @@ func parent(leafIndex int64) int64 {
 func isRightChild(leafIndex int64) bool {
 	return leafIndex&1 == 1
 }
+
+// bit returns the i'th bit of index from the right.
```

Comment:
I think index 0 being rightmost bit is OK as it matches std::bitset in C++. Can you add an example for increased clarity? Preferably a multi-byte one. For example if there's 2 bytes of input bit 7 is the MSB of index[1] and bit 8 is the LSB of index[0]?